pallets/lbp/src/lib.rs

```diff
@@ -658,7 +663,13 @@ pub mod pallet {
 
 			Self::deposit_event(Event::LiquidityRemoved(pool_id, asset_a, asset_b, amount_a, amount_b));
 
-			Ok(())
+			 let amount_a_perbill: Balance = transfer_to_xyk * amount_a;
```

Comment:
calculate only if perbill is > 0

Reply:
We are testing the value after multiplication because small non zero values of Perbill can produce zero values. This way we are sure that we work with non zero values only. But I think we can move this check to the `T::OnRemoveLiquidity::on_end_of_lbp` and throw some error from there if the values are zero.

Reply:
i dont think that transfer of 0 tokens is concern for the LBP pallet